### PR TITLE
Verify GPU Direct RDMA is used on supported instance.

### PR DESCRIPTION
**Pull request by weicongw**
*Issue #, if available:*

*Description of changes:*
- Verify GPU Direct RDMA is used on supported instance.
- Update the nvidia and efa plugin to the latest.
- Correct the persistence status of P5 instance.

*Testing*

NCCL test on bad AMI: 
```
...
        [1,9]<stdout>:multi-node-nccl-test-worker-1:21:69 [1] NCCL INFO Channel 01/0 : 9[1] -> 8[0] via P2P/IPC
        [1,8]<stdout>:multi-node-nccl-test-worker-1:20:67 [0] NCCL INFO Channel 00/0 : 8[0] -> 0[0] [send] via NET/Socket/0
...
        [1,10]<stderr>:libfabric:22:1725400706::efa:domain:efa_domain_hmem_info_init_cuda():169<warn> Failed to register CUDA buffer with the EFA device, FI_HMEM transfers that require peer to peer support will fail.
...
        [1,0]<stdout>:multi-node-nccl-test-worker-0:20:20 [0] NCCL INFO comm 0x55a496b65e90 rank 0 nranks 16 cudaDev 0 busId 53000 - Destroy COMPLETE
        [1,0]<stdout>:# Out of bounds values : 0 OK
        [1,0]<stdout>:# Avg bus bandwidth    : 1.00991 
        [1,0]<stdout>:#
...     
    mpi_test.go:137: GPU Direct RDMA is not utilized for inter-node communication in NCCL tests on instances that support GDRDMA: p5.48xlarge
--- FAIL: TestMPIJobPytorchTraining (751.57s)
    --- SKIP: TestMPIJobPytorchTraining/single-node (0.00s)
    --- FAIL: TestMPIJobPytorchTraining/multi-node (751.57s)
        --- FAIL: TestMPIJobPytorchTraining/multi-node/MPIJob_succeeds (750.79s)
```
NCCL test on good AMI: 
```
        [1,3]<stdout>:multi-node-nccl-test-worker-0:24:79 [3] NCCL INFO NET/OFI Libfabric provider associates MRs with domains
        [1,7]<stdout>:multi-node-nccl-test-worker-0:33:74 [7] NCCL INFO Channel 09/0 : 7[7] -> 15[7] [send] via NET/AWS Libfabric/7/GDRDMA
...
        [1,0]<stdout>:# Out of bounds values : 0 OK
        [1,0]<stdout>:# Avg bus bandwidth    : 21.2232 
        [1,0]<stdout>:#
...
        
--- PASS: TestMPIJobPytorchTraining (271.88s)
    --- SKIP: TestMPIJobPytorchTraining/single-node (0.00s)
    --- PASS: TestMPIJobPytorchTraining/multi-node (271.88s)
        --- PASS: TestMPIJobPytorchTraining/multi-node/MPIJob_succeeds (271.09s)
```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: e2e2/test/cases/nvidia/manifests/efa-device-plugin.yaml [hunk 1]
```diff
@@ -90,7 +155,16 @@ spec:
           volumeMounts:
             - name: device-plugin
               mountPath: /var/lib/kubelet/device-plugins
+            - name: infiniband-volume
+              mountPath: /dev/infiniband
+          resources:
+            requests:
+              cpu:    10m
+              memory: 20Mi
```
Comment (Issacwww) [inline e2e2/test/cases/nvidia/manifests/efa-device-plugin.yaml:163]: what is this for?

File: e2e2/test/cases/nvidia/manifests/efa-device-plugin.yaml [hunk 2]
```diff
@@ -24,63 +22,130 @@ spec:
...
+                    - hpc7a.48xlarge
+                    - hpc7a.96xlarge
+                    - hpc7g.4xlarge
+                    - hpc7g.8xlarge
+                    - hpc7g.16xlarge
       hostNetwork: true
       containers:
-        - image: 602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/aws-efa-k8s-device-plugin:v0.3.3
-          imagePullPolicy: Always
+        - image: 602401143452.dkr.ecr.us-west-2.amazonaws.com/eks/aws-efa-k8s-device-plugin:v0.5.0
```
Comment (Issacwww) [inline e2e2/test/cases/nvidia/manifests/efa-device-plugin.yaml:148]: is there a latest tag for us to use so we can avoid keep updating version?

File: e2e2/test/cases/nvidia/manifests/nvidia-device-plugin.yaml [hunk 3]
```diff
@@ -38,7 +40,7 @@ spec:
       # See https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/
       priorityClassName: "system-node-critical"
       containers:
-      - image: nvcr.io/nvidia/k8s-device-plugin:v0.14.2
+      - image: nvcr.io/nvidia/k8s-device-plugin:v0.16.2
```
Comment (Issacwww) [inline e2e2/test/cases/nvidia/manifests/nvidia-device-plugin.yaml:43]: same here, is there latest tag?

Comment (weicongw) [reply e2e2/test/cases/nvidia/manifests/nvidia-device-plugin.yaml:43]: No they don't

File: e2e2/test/cases/nvidia/manifests/efa-device-plugin.yaml (hunk 1 above)
Comment (weicongw) [reply e2e2/test/cases/nvidia/manifests/efa-device-plugin.yaml:163]: It is synced from https://github.com/aws-samples/aws-efa-eks/blob/main/manifest/efa-k8s-device-plugin.yml

File: e2e2/test/cases/nvidia/manifests/efa-device-plugin.yaml (hunk 2 above)
Comment (weicongw) [reply e2e2/test/cases/nvidia/manifests/efa-device-plugin.yaml:148]: Confirmed with EFA team, there is no latest tag.

**Review by Issacwww — APPROVED** [review]

File: e2e2/test/cases/nvidia/manifests/efa-device-plugin.yaml (hunk 1 above)
Comment (bryantbiggs) [reply e2e2/test/cases/nvidia/manifests/efa-device-plugin.yaml:163]: we should be using this https://github.com/aws/eks-charts/tree/master/stable/aws-efa-k8s-device-plugin

**Pull request merged**